cymetric/convenient_interface.py

```diff
@@ -0,0 +1,650 @@
+    group_start = group_end + ['Mass']
+    df = df[group_start].groupby(group_end).sum()
+    df.reset_index(inplace=True)
+
+    agents_list = evaler_.eval('AgentEntry')['Prototype'].tolist()
+
+    dot = Digraph('G')
+
+    for agent in agents_list:
+        dot.node(agent)
+
+    for index, row in df.iterrows():
+        dot.edge(row['SenderPrototype'], row['ReceiverPrototype'],
+                 label=str(row['Mass']))
+
+    return dot
+
+
+def get_inventory_df(evaler_, fac_list=[], nuc_list=[]):
+    """
+    Shape the reduced inventory Data Frame. Applying nuclides/facilities selection when required.
+
+    Parameters
+    ----------
+    evaler_: evaler
+    fac_list:  list of the facility
+    nuc_list: list of nuclide to select.
+    """
+
+    # Get inventory table
+    df = evaler_.eval('ExplicitInventory')
+    agents = evaler_.eval('AgentEntry')
+
+    rdc_list = []  # because we want to get rid of the nuclide asap
+    if len(nuc_list) != 0:
+        nuc_list = format_nuclist(nuc_list)
+        rdc_list.append(['NucId', nuc_list])
+
+    if len(fac_list) != 0:
+        agents = agents[agents['Prototype'].isin(fac_list)]
+        rdc_list.append(['AgentId', agents['AgentId'].tolist()])
+    else:
+        wng_msg = "no faciity provided"
+        warnings.warn(wng_msg, UserWarning)
+    df = reduce(df, rdc_list)
+
+    base_col = ['SimId', 'AgentId']
+    added_col = base_col + ['Prototype']
+    df = merge(df, base_col, agents, added_col)
+
+    return df
+
+
+def get_inventory_timeserie(evaler_, fac_list=[], nuc_list=[]):
+    """
+    Shape the reduced inventory Data Frame into a simple time serie. Applying
+    nuclides/facilities selection when required.
+
+    Parameters
+    ----------
+    evaler_: evaler
+    fac_list:  list of the facility
+    nuc_list: list of nuclide to select.
+    """
+
+    if len(nuc_list) != 0:
+        nuc_list = format_nuclist(nuc_list)
+    else:
+        wng_msg = "no nuclide provided"
+        warnings.warn(wng_msg, UserWarning)
+
+    df = get_inventory_df(evaler_, fac_list, nuc_list)
+
+    group_end = ['Time']
+    group_start = group_end + ['Quantity']
+    df = df[group_start].groupby(group_end).sum()
+    df.reset_index(inplace=True)
+
+    time = evaler_.eval('TimeList')
+    df = add_missing_time_step(df, time)
+    return df
+
+
+def get_inventory_activity_df(evaler_, fac_list=[], nuc_list=[]):
+    """
+    Get a simple time series of the activity of the inventory in the selcted
+    facilities. Applying nuclides selection when required.
+
+    Parameters
+    ----------
+    evaler_: evaler
+    fac_list:  list of the facility
+    nuc_list: list of nuclide to select.
+    """
+
+    if len(nuc_list) != 0:
+        nuc_list = format_nuclist(nuc_list)
+
+    df = get_inventory_df(evaler_, fac_list, nuc_list)
+    for i, row in df.iterrows():
+        val = 1000 * data.N_A * df.ix[i, 'Quantity'] * \
+            data.decay_const(int(df.ix[i, 'NucId']))
```

Comment:
if you have the row, you shouldn't need to look the index back up on the dataframe right?